java/system-test/src/main/java/sleeper/systemtest/ingest/WriteRandomDataViaQueueJob.java

```diff
@@ -107,25 +109,28 @@ public void run() throws IOException {
                     .build();
             String jsonJob = new IngestJobSerDe().toJson(ingestJob);
             LOGGER.debug("Sending message to ingest queue ({})", jsonJob);
-            sendMessageRequest = new SendMessageRequest()
+            sendMessageRequests.add(new SendMessageRequest()
                     .withQueueUrl(getSystemTestProperties().get(INGEST_JOB_QUEUE_URL))
-                    .withMessageBody(jsonJob);
+                    .withMessageBody(jsonJob));
         } else if (ingestMode.equalsIgnoreCase(IngestMode.BULK_IMPORT_QUEUE.name())) {
-            BulkImportJob bulkImportJob = new BulkImportJob.Builder()
-                    .tableName(getTableProperties().get(TABLE_NAME))
-                    .id(UUID.randomUUID().toString())
-                    .files(Arrays.asList(dir))
-                    .build();
-            String jsonJob = new BulkImportJobSerDe().toJson(bulkImportJob);
-            LOGGER.debug("Sending message to ingest queue ({})", jsonJob);
-            sendMessageRequest = new SendMessageRequest()
-                    .withQueueUrl(getSystemTestProperties().get(BULK_IMPORT_EMR_JOB_QUEUE_URL))
-                    .withMessageBody(jsonJob);
+            int totalBulkImportJobs = getSystemTestProperties().getInt(NUMBER_OF_BULK_IMPORT_JOBS);
+            for (int i = 0; i < totalBulkImportJobs; i++) {
+                BulkImportJob bulkImportJob = new BulkImportJob.Builder()
+                        .tableName(getTableProperties().get(TABLE_NAME))
+                        .id(UUID.randomUUID().toString())
+                        .files(List.of(dir))
+                        .build();
+                String jsonJob = new BulkImportJobSerDe().toJson(bulkImportJob);
+                LOGGER.debug("Sending message to ingest queue ({})", jsonJob);
+                sendMessageRequests.add(new SendMessageRequest()
+                        .withQueueUrl(getSystemTestProperties().get(BULK_IMPORT_EMR_JOB_QUEUE_URL))
+                        .withMessageBody(jsonJob));
+            }
         } else {
             throw new IllegalArgumentException("Unknown ingest mode of " + ingestMode);
         }
 
-        sqsClient.sendMessage(sendMessageRequest);
+        sendMessageRequests.forEach(sqsClient::sendMessage);
```

Comment:
The idea was to create one bulk import job for the output of all the data generation tasks. This class runs in each data generation task, so each task will submit multiple jobs.

It looks like we need a separate ingest mode to aggregate the output of the data generation tasks into a job. We could have a separate mode to say that we'll handle the actual ingest separately and the tasks are just to generate the data. An alternative would be to rework it to always separate the actual ingest from the data generation.

It's also a bit misleading that there's logic in here to split the records into multiple files, but in practice it will only ever generate one file because we generate less than 100 million records in each task.

Reply:
We've got a couple of options to handle this:

1. Add an ingest mode that says just generate the data and don't actually ingest. The test scripts will handle building the ingest job.
2. Create a Lambda that will poll the state of the system, and will gather the output from the data generation tasks and create the ingest jobs. This Lambda could also be used to execute other steps of the test that are currently invoked by scripts. It would need to examine the current state of the system and run the next step in the test when it is time. We could store the state of the test in S3.

Reply:
The Lambda-based approach would be useful for performance tests, but I think ideally we'd like to be able to use JUnit for system tests. That has the disadvantage of needing to run on a development environment with Maven, but that's probably worthwhile if it's automated and we don't need to sit and wait for it. We can limit the amount of time we spend on any approach to system tests that doesn't work that way.